src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,33 @@
+package duke;
+
+/**
+ * The unique exceptions thrown by Duke.
+ */
+public class DukeException extends Exception {
+    protected String description;
+
+    /**
+     * Constructor for DukeException for an invalid input.
+     */
+    public DukeException() {
+        super();
+        this.description = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    }
+
+    /**
+     * Constructor for DukeException for a missing task descriptor.
```

Comment:
Minor nit, but I think there should be an empty line between the descriptor and params sections here? I noticed that you added them for some other comments, so maybe can be more consistent with the javadocs comments.